### PR TITLE
FOGL-8754 SHA512 for login token and user management passwords

### PR DESCRIPTION
**Pull request by ashish-jabble**

### CODE REVIEW

File: scripts/plugins/storage/sqlite/upgrade/71.sql [hunk 1]
```diff
@@ -0,0 +1,5 @@
+-- Add new column name 'hash_algorithm' in users table
+
+ALTER TABLE fledge.users ADD COLUMN hash_algorithm TEXT CHECK( hash_algorithm IN ('SHA256', 'SHA512') )  NOT NULL DEFAULT 'SHA512';
+UPDATE fledge.users SET hash_algorithm='SHA256';
+UPDATE fledge.users SET pwd='495f7f5b17c534dbeabab3da2287a934b32ed6876568563b04c312be49e8773299243abd3881d13112ccfb67c4fb3ec8231406474810e1f6eb347d61c63785d4:672169c60df24b76b6b94e78cad800f8', hash_algorithm='SHA512' WHERE pwd ='39b16499c9311734c595e735cffb5d76ddffb2ebf8cf4313ee869525a9fa2c20:f400c843413d4c81abcba8f571e6ddb6';
```
Comment (praveen-garg) [inline scripts/plugins/storage/sqlite/upgrade/71.sql:5]: If we are setting SHA256 for existing then no need to change defaults

File: python/fledge/common/web/middleware.py [hunk 2]
```diff
@@ -104,6 +104,8 @@ async def middleware(request):
                 raise web.HTTPUnauthorized(reason=e)
             except (jwt.DecodeError, jwt.ExpiredSignatureError, User.InvalidToken, User.TokenExpired) as e:
                 raise web.HTTPUnauthorized(reason=e)
+            except jwt.exceptions.InvalidAlgorithmError:
+                raise web.HTTPUnauthorized(reason="The token has expired, login again.")
```
Comment (praveen-garg) [inline python/fledge/common/web/middleware.py:108]: log in

File: scripts/plugins/storage/sqlite/upgrade/71.sql (hunk 1 above)
Comment (ashish-jabble) [reply scripts/plugins/storage/sqlite/upgrade/71.sql:5]: > we are setting SHA256
SHA256 for old users only; as in schema by default SHA512

No harm to update the default users with SHA512 and good to have

Comment (praveen-garg) [reply scripts/plugins/storage/sqlite/upgrade/71.sql:5]: Sure

Are we ensuring that on reset or change password, SHA512 gets effective?

Comment (ashish-jabble) [reply scripts/plugins/storage/sqlite/upgrade/71.sql:5]: yeah

**Review by praveen-garg — APPROVED** [review]

**Review by MarkRiddoch — APPROVED** [review]

**Pull request merged**